### PR TITLE
chore(test): add combineLatest virtual time tests from RxJS 2/3

### PR DESCRIPTION
**Pull request by blesh**
- adds virtual time tests from RxJS2/3 with marbley goodness
- updates TestScheduler and test-helpers a bit for better testing.

### CODE REVIEW

**Comment by staltz** [comment]
<3

File: spec/operators/combineLatest-spec.js [hunk 1]
```diff
@@ -67,4 +70,254 @@ describe('combineLatest', function () {
...
+  it('should work with hot-empty and hot-single', function () {    
+    var values = { 
+      a: 1, 
+      b: 2, 
+      c: 3, 
+      r: 1 + 3 //a + c
+    };
+    var e1 =        hot('-a-^-|', values);
+    var e2 =        hot('-b-^-c-|', values);
+    var expected =         '----|';
```
Comment (staltz) [inline spec/operators/combineLatest-spec.js]: Where are these `hot()` helpers actually actually implemented?

Comment (blesh) [inline spec/operators/combineLatest-spec.js]: There's a test-helpers.js file in the spec directory under helpers (I think). I touched it in this PR.

Comment (blesh) [inline spec/operators/combineLatest-spec.js]: spec/helpers/test-helper.js

Comment (blesh) [inline spec/operators/combineLatest-spec.js]: They're really on TestScheduler, though

**Pull request merged**